common/nym-lp/src/session.rs

```diff
@@ -29,11 +128,71 @@ pub struct LpSession {
     /// Noise protocol state machine
     noise_state: Mutex<NoiseProtocol>,
 
+    /// KKT (KEM Key Transfer) exchange state
+    kkt_state: Mutex<KKTState>,
+
+    /// PSQ (Post-Quantum Secure PSK) handshake state
+    psq_state: Mutex<PSQState>,
+
+    /// PSK handle from responder (ctxt_B) for future re-registration
+    psk_handle: Mutex<Option<Vec<u8>>>,
```

Comment:
it doesn't seem like all of those have to actually be behind mutexes, what about using a cheaper wrapper instead, like a `Cell`?